src/r_util.cpp

```diff
@@ -487,6 +488,22 @@ namespace rhost {
             return Rf_ScalarInteger(static_cast<int>(lock_state));
         }
 
+        extern "C" SEXP fetch_file(SEXP path) {
+            fs::path fpath;
+            r_top_level_exec([&]() {
+                fpath = fs::u8path(Rf_translateCharUTF8(STRING_ELT(path, 0)));
+            }, __FUNCTION__);
+
+            if (!fpath.empty()) {
+                blobs::blob file_data;
+                blobs::append_from_file(file_data, fpath);
+                host::send_notification("!FetchFile", file_data, fpath.filename().string());
```

Comment:
Can you elaborate? They way I see it, this is no different from plot. User invokes a function, and RTVS has to handle it in a unique manner. The contents of the notification, in this case have to be saved. 

---
In reply to: [73976559](https://github.com/Microsoft/R-Host/pull/77#discussion_r73976559) [](ancestors = 73976559)